### PR TITLE
JWT, login page & routes middleware

### PR DESCRIPTION
**Pull request by CKYWEB**

### CODE REVIEW

File: app/api/_services/user.ts [hunk 1]
```diff
@@ -25,11 +26,17 @@ interface UserProps extends Omit<IUser, "_id" | "profileId" | "rolesId"> {
...
-    return false;
+  try {
+    const user = await getUser({ email }, ["email", "password"]);
+
+    if (!user || !bcrypt.compareSync(password, user.password)) {
+      return false;
+    }
+
+    return await user.toObject();
+  } catch (e) {
```
Comment (INEEDAMONITOR) [inline app/api/_services/user.ts:37]: Why return the user object instead of a boolean?

File: app/api/_services/user.ts [hunk 2]
```diff
@@ -76,10 +82,17 @@ export const createUser = async (user: CreateUserProp) => {
...
       ownerId: newUser._id,
     }).exec();
-    // TODO: Return the JWT token
+
+    await sendbirdRequests.createUser({
+      user_id: newRole._id,
+      nickname: newProfile.displayName,
+      profile_url: "https://sendbird.com/main/img/profiles/profile_05_512px.png",
+      issue_access_token: true,
+    });
```
Comment (INEEDAMONITOR) [inline app/api/_services/user.ts:95]: It seems like you created a sendbird user during the Cearte User. But the sendbird user should be linked to the Role. As the role is the source for the chat.

File: app/api/login/route.ts [hunk 3]
```diff
@@ -1,39 +1,67 @@
...
+        {
+          status: 401
+        }
+      );
+    }
+    delete user.password;
+
+    const token = jwt.sign(user, process.env.JWT_SECRET as string, {
+      expiresIn: 60 * 60 * 24 * Number(process.env.JWT_EXPIRES_IN),
+    });
```
Comment (INEEDAMONITOR) [inline app/api/login/route.ts:47]: Might need to featch roles if rolesId remove from user model

**Review by INEEDAMONITOR — APPROVED** [review]

File: app/api/_services/user.ts (hunk 1 above)
Comment (CKYWEB) [reply app/api/_services/user.ts:37]: The user data needs to be retained for subsequent use in the JWT signing process. See: https://github.com/INEEDAMONITOR/Role-Rally/blob/main/app/api/login/route.ts

File: app/api/_services/user.ts (hunk 2 above)
Comment (CKYWEB) [reply app/api/_services/user.ts:95]: The user creation function automatically assigns a default role. Therefore, I think it is advisable to create a new Sendbird user using the ID of the newly assigned role.

File: app/api/_middleware/user.ts
Comment (INEEDAMONITOR) [inline app/api/_middleware/user.ts:1]: link that might useful: [middleware for next js](https://medium.com/sopra-steria-norge/how-to-write-actual-api-middleware-for-next-js-2a38355f6674)

**Pull request merged**